leptonai/api/types.py

```diff
@@ -165,21 +165,25 @@ def make_env_vars_from_strings(
                 k, v = s.split("=", 1)
             except ValueError:
                 raise ValueError(f"Invalid environment definition: [red]{s}[/]")
-            if k.lower().startswith(LEPTON_RESERVED_ENV_PREFIX):
+            if k in LEPTON_RESERVED_ENV_NAMES:
```

Comment:
do we wanna check `k.upper()`?

Reply:
Strictly speaking environmental variables are case sensitive, so one can indeed have "leptonAI_workspace_ID" and still not have conflicts... so I kept it this way to be minimally restrictive.